gamemode/modules/commands/sv_commands.lua

```diff
@@ -0,0 +1,260 @@
+util.AddNetworkString( "RVR_Commands_runConsoleCommand" )
+
+RVR.Commands = RVR.Commands or {}
+local commands = RVR.Commands
+
+commands.list = {}
+commands.types = {}
+
+function commands.addType( name, func )
+    if commands.types[name] then
+        error( "A type with the same name (" .. name .. ") already exists" )
+    end
+
+    commands.types[name] = func
+end
+
+commands.addType( "int", function( arg )
+    local num = tonumber( arg )
+
+    if not num or num % 1 ~= 0 then
+        return nil, "Invalid integer: " .. arg
+    end
+
+    return num
+end )
+
+commands.addType( "float", function( arg )
+    local num = tonumber( arg )
+
+    if not num then
+        return "Invalid float: " .. arg
+    end
+
+    return num
+end )
+
+local booleanValues = {
+    ["enabled"]  = true,
+    ["enable"]   = true,
+    ["true"]     = true,
+    ["yes"]      = true,
+    ["1"]        = true,
+
+    ["disabled"] = false,
+    ["disable"]  = false,
+    ["false"]    = false,
+    ["no"]       = false,
+    ["0"]        = false
+}
+
+commands.addType( "bool", function( arg )
+    if booleanValues[arg] ~= nil then
+        return booleanValues[arg]
+    end
+
+    return nil, "Invalid boolean: " .. arg
+end )
+
+commands.addType( "string", function( arg )
+    return arg
+end )
+
+commands.addType( "player", function( arg )
+    local isSteamID = string.find( arg, "STEAM_" )
+
+    if isSteamID then
+        local ply = player.GetBySteamID( arg )
+
+        if ply then
+            return ply
+        else
+            return nil, "Invalid player: " .. arg
+        end
+    end
+
+    for _, ply in pairs( player.GetAll() ) do
+        if string.find( ply:Nick(), arg ) then return ply end
+    end
+
+    return nil, "Invalid player: " .. arg
+end )
+
+function commands.checkArguments( argNames, argTypes, args )
+    if #args < #argTypes then
+        local commandHelp = ""
+
+        for i, argName in ipairs( argNames ) do
+            local argType = argTypes[i]
+
+            commandHelp = commandHelp .. argName .. ":" .. argType
+        end
+
+        return nil, "Missing argument: " .. commandHelp
```

Comment:
```suggestion
        return nil, "Missing argument. Usage: " .. commandHelp
```
Doing "Missing argument: " implies the next thing will be the argument that is missing.